packages/runtime/container-runtime/src/test/containerRuntime.spec.ts

```diff
@@ -2644,6 +2661,82 @@ describe("Runtime", () => {
 					"SignalLost and SignalOutOfOrder telemetry should not be logged when signals are processed in order",
 				);
 			});
+
+			it("increments signalLost field after successive dropped signals", () => {
+				// Send 5 signals
+				sendSignals(5);
+
+				// Process the first signal
+				processSignals(submittedSignals, 1);
+
+				// Drop the second signal
+				dropSignals(1);
+
+				// Process the third signal
+				processSignals(submittedSignals, 1);
+
+				// Missing second signal should be detected
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLost",
+							signalsLost: 1,
+						},
+					],
+					"SignalLost telemetry should be logged when signal is dropped",
+				);
+
+				// Drop the fourth signal
+				dropSignals(1);
+
+				// Process the fifth signal
+				processSignals(submittedSignals, 1);
+
+				// Missing fourth signal should be detected
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLost",
+							signalsLost: 2,
```

Comment:
Hmm. If we are going to do a cumulative signal lost count, then there should also be a relative signal lost count. Telemetry can be lossy. I expect we care more about how many signals just went lost versus the total count.
If we only had the relative count, we could simply sum up all of the relative counts (and get reasonable numbers even if a little is lost).

Reply:
Ok that makes sense. I updated the logic and corresponding tests to only report relative lost signal count once detected.